android/src/main/java/com/klarna/mobile/sdk/reactnative/signin/KlarnaSignInModuleImpl.java

```diff
@@ -0,0 +1,160 @@
+
+import androidx.annotation.NonNull;
+
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.klarna.mobile.sdk.KlarnaMobileSDKError;
+import com.klarna.mobile.sdk.api.KlarnaEnvironment;
+import com.klarna.mobile.sdk.api.KlarnaEventHandler;
+import com.klarna.mobile.sdk.api.KlarnaProductEvent;
+import com.klarna.mobile.sdk.api.KlarnaRegion;
+import com.klarna.mobile.sdk.api.component.KlarnaComponent;
+import com.klarna.mobile.sdk.api.signin.KlarnaSignInEvent;
+import com.klarna.mobile.sdk.api.signin.KlarnaSignInSDK;
+import com.klarna.mobile.sdk.reactnative.common.util.ArgumentsUtil;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class KlarnaSignInModuleImpl implements KlarnaEventHandler {
+
+    private static final String PARAM_NAME_ACTION = "action";
+    private static final String PARAM_NAME_KLARNA_MESSAGE_EVENT = "klarnaMessageEvent";
+    private static final String PARAM_NAME_PARAMS = "params";
+
+    public static final String NAME = "RNKlarnaSignIn";
+
+    private final ReactApplicationContext reactAppContext;
+    private final ArrayList<KlarnaSignInData> signInSDKList;
+
+    public KlarnaSignInModuleImpl(ReactApplicationContext reactAppContext) {
+        this.reactAppContext = reactAppContext;
+        this.signInSDKList = new ArrayList<KlarnaSignInData>();
+    }
+
+    /* Module private methods */
+
+    private KlarnaEnvironment environmentFrom(@NonNull String value) {
+        return switch (value) {
+            case "playground" -> KlarnaEnvironment.PLAYGROUND;
+            case "staging" -> KlarnaEnvironment.STAGING;
+            default -> KlarnaEnvironment.PRODUCTION;
+        };
+
+    }
+
+    private KlarnaRegion regionFrom(@NonNull String value) {
+        return switch (value) {
+            case "na" -> KlarnaRegion.NA;
+            case "oc" -> KlarnaRegion.OC;
+            default -> KlarnaRegion.EU;
+        };
+
+    }
+
+    private KlarnaSignInData getInstanceData(KlarnaComponent component) {
+        for (KlarnaSignInData data : signInSDKList) {
+            if (data.sdkInstance == component) {
+                return data;
+            }
+        }
+        return null;
+    }
+
+    private KlarnaSignInData getInstanceData(String instanceId) {
+        for (KlarnaSignInData data : signInSDKList) {
+            if (data.instanceId.equals(instanceId)) {
+                return data;
+            }
+        }
+        return null;
+    }
+
+    private String getParamsFrom(KlarnaProductEvent event) {
+        String paramsJson = "{}";
+        try {
+            paramsJson = gson.toJson(event.getParams());
+        } catch (Exception ignored) {
+        }
+        return paramsJson;
+    }
+
+    /* Module public methods */
+    public void init(String instanceId, String environment, String region, String returnUrl) {
```

Comment:
should we receive a promise here too to resolve? in case something fails in the init block we can reject it too